tests/e2e/s/simplex.go

```diff
@@ -0,0 +1,197 @@
+	leader := getLeaderForRound(nodes, round)
+	tc.Log().Info("current height and round",
+		zap.Uint64("height", latestBlock.Height),
+		zap.Uint64("round", round),
+		zap.Stringer("leader", leader.NodeID),
+	)
+
+	// record the nonces of all nodes before issuing the tx
+	nonces := make(map[ids.NodeID]uint64)
+	for _, node := range nodes {
+		client = api.NewClient(node.GetAccessibleURI(), chain.ChainID.String())
+		nonce, err := client.Nonce(tc.DefaultContext(), chain.PreFundedKey.Address())
+		require.NoError(err)
+		nonces[node.NodeID] = nonce
+	}
+
+	// issue a transfer transaction to the leader
+	leaderURI := leader.GetAccessibleURI()
+	tc.Log().Info("issuing XSVM transfer transaction for simplex subnet",
+		zap.Stringer("leader", leader.NodeID),
+		zap.String("leaderURI", leaderURI),
+	)
+	recipientKey := e2e.NewPrivateKey(tc)
+
+	// create and sign the transaction
+	utx := &tx.Transfer{
+		ChainID: chain.ChainID,
+		Nonce:   nonces[leader.NodeID],
+		MaxFee:  0,
+		AssetID: chain.ChainID,
+		Amount:  units.Schmeckle,
+		To:      recipientKey.Address(),
+	}
+	stx, err := tx.Sign(utx, chain.PreFundedKey)
+	require.NoError(err)
+	txID, err := stx.ID()
+	require.NoError(err)
+
+	// issue txs to all nodes, because XSVM does not have mempool gossiping
+	for i, node := range nodes {
+		client = api.NewClient(node.GetAccessibleURI(), chain.ChainID.String())
+		_, err := client.IssueTx(tc.DefaultContext(), stx)
+		require.NoError(err, "node %d failed to issue tx", i)
+	}
+
+	tc.Log().Info("successfully issued XSVM transfer transactions",
+		zap.Stringer("txID", txID),
+		zap.Uint64("round", round),
+	)
+
+	tc.By("checking all nodes have accepted the tx")
+	address := chain.PreFundedKey.Address()
+	for _, node := range nodes {
+		client = api.NewClient(node.GetAccessibleURI(), chain.ChainID.String())
+		err = api.AwaitTxAccepted(tc.DefaultContext(), client, address, nonces[node.NodeID], api.DefaultPollingInterval)
+		require.NoError(err, "node %s failed to accept tx in round %d", node.NodeID, round)
+
+		balance, err := client.Balance(tc.DefaultContext(), recipientKey.Address(), chain.ChainID)
+		require.NoError(err)
+		require.Equal(units.Schmeckle, balance)
+
+		_, statelessBlock, err := client.LastAccepted(tc.DefaultContext())
+		require.NoError(err)
+
+		require.Len(statelessBlock.Txs, 1)
+		require.Equal(statelessBlock.Txs[0], stx)
+
+		tc.Log().Info("node has accepted the tx",
+			zap.Stringer("node", node.NodeID),
+			zap.Stringer("txID", txID),
+			zap.Uint64("height", statelessBlock.Height),
+			zap.Uint64("round", round),
+		)
+	}
+}
+
+// Retrieve the nodes corresponding to the provided IDs
+func getNodesForIDs(nodes []*tmpnet.Node, nodeIDs []ids.NodeID) []*tmpnet.Node {
+	desiredNodes := make([]*tmpnet.Node, 0, len(nodeIDs))
+	for _, node := range nodes {
+		for _, nodeID := range nodeIDs {
+			if node.NodeID == nodeID {
+				desiredNodes = append(desiredNodes, node)
+			}
+		}
+	}
+	return desiredNodes
+}
```

Comment:
Since it is in tests, it doesn't actually matter... But we typically use sets for filtering
```suggestion
func getNodesWithIDs(nodes []*tmpnet.Node, nodeIDs set.Set[ids.NodeID]) []*tmpnet.Node {
	desiredNodes := make([]*tmpnet.Node, 0, len(nodeIDs))
	for _, node := range nodes {
		if nodeIDs.Contains(node.NodeID) {
			desiredNodes = append(desiredNodes, node)
		}
	}
	return desiredNodes
}
```